R/derive_param_lasta.R

```diff
@@ -0,0 +1,358 @@
+
+  }
+
+  # Error if filter results in 0 records
+  if (nrow(dataset_filter) == 0) {
+    err_msg <- sprintf(
+      "dataframe passed into %s argument with the filter %s has 0 records",
+      "dataset",
+      deparse(rlang::quo_get_expr(filter_source)))
+
+    rlang::abort(err_msg)
+  }
+
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  # filter_extreme: Filter last assessment using filter_extreme ----
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  param_lasta_01 <- dataset_filter %>%
+      admiral::filter_extreme(mode       = "last",
+                              order      = admiral::vars(!!!order),
+                              by_vars    = admiral::vars(!!!by_vars),
+                              check_type = "warning")
+
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  # adsl: Append subjects in ADSL not in dataset_filter ----
+  #
+  # Note Requirement: For subjects without observations in the input dataset
+  # after the filter is applied, we keep all columns from ADSL which
+  # are also in the input dataset.
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  # create list of unique subject_keys with at least one record
+  subjects_with_data <-  dataset_filter %>%
+    dplyr::distinct(!!!subject_keys)
+
+  # All records in dataset_adsl that do not have a match in subjects_with_data are
+  # removed with anti_join.
+  adsl_subjects_only <- dataset_adsl %>%
+    dplyr::anti_join(subjects_with_data,
+                     by = admiral::vars2chr(subject_keys)) %>%
+    dplyr::select(dplyr::intersect(colnames(dataset_adsl),
+                                   colnames(dataset)))
+
+  # bind back with dataframe with data
+  param_lasta_02 <- param_lasta_01 %>%
+    dplyr::bind_rows(adsl_subjects_only)
+
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  # set_values_to: Execute set_values_to ----
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  tryCatch(
+
+    param_lasta_values_set <- param_lasta_02 %>%
+      dplyr::mutate(!!!set_values_to),
+
+    error = function(cnd) {
+      abort(
+        paste0(
+          "Assigning new columns with set_values_to has failed:\n",
+          "set_values_to = (\n",
+          paste(
+            " ",
+            names(set_values_to),
+            "=",
+            lapply(set_values_to, rlang::quo_get_expr),
+            collapse = "\n"
+          ),
+          "\n)\nError message:\n  ",
+          cnd
+        )
+      )
+    })
+
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  # Bind back to passed dataset and return ----
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  return_dataframe <- dplyr::bind_rows(dataset,
+                                       param_lasta_values_set)
+
+  return(return_dataframe)
```

Comment:
Understood, I will update, though I am not a big fan of tidyverse style convention.  Funnily enough styler does not pick up on this.

Reply:
Please do not assign the final data frame to a variable. Otherwise, the value is returned silently, i.e., the result is not displayed if the function is called without assigning the result to a variable.